shipyard-controller/handler/shipyardcontroller.go

```diff
@@ -575,16 +551,30 @@ var errNoFurtherTaskForSequence = errors.New("no further task for sequence")
 var errNoTaskSequence = errors.New("no task sequence found")
 var errNoStage = errors.New("no stage found")
 
-func (sc *shipyardController) getTaskSequencesByTrigger(eventScope *keptnv2.EventData, completedTaskSequence string, shipyard *keptnv2.Shipyard) []NextTaskSequence {
+func getTaskSequencesByTrigger(eventScope *keptnv2.EventData, completedTaskSequence string, shipyard *keptnv2.Shipyard) []NextTaskSequence {
 	result := []NextTaskSequence{}
 	for _, stage := range shipyard.Spec.Stages {
 		for tsIndex, taskSequence := range stage.Sequences {
-			for _, trigger := range taskSequence.Triggers {
-				if trigger == eventScope.Stage+"."+completedTaskSequence+".finished" {
-					result = append(result, NextTaskSequence{
-						Sequence:  stage.Sequences[tsIndex],
-						StageName: stage.Name,
-					})
+			for _, trigger := range taskSequence.TriggeredOn {
+				if trigger.Event == eventScope.Stage+"."+completedTaskSequence+".finished" {
+					appendSequence := false
+					// default behavior if no selector is available: 'pass', as well as 'warning' results trigger this sequence
+					if trigger.Selector.Match == nil {
+						if eventScope.Result == keptnv2.ResultPass || eventScope.Result == keptnv2.ResultWarning {
+							appendSequence = true
+						}
+					} else {
+						// if a selector is there, compare the 'result' property
+						if string(eventScope.Result) == trigger.Selector.Match["result"] {
```

Comment:
would it make sense to also allow matching against the other top level fields?

Reply:
yes, but for now we agreed in https://github.com/keptn/keptn/issues/3028 to excplicitly check for `result` for now.